proxy/proxy-server.py

```
@@ -0,0 +1,69 @@
+
+        if SECRET and body.get("secret") != SECRET:
+            self.send_error(401, "Unauthorized")
+            return
+
+        url = body.get("url", "")
+        if not url or not is_allowed(url):
+            self.send_error(400, "Invalid or disallowed URL")
+            return
+
+        headers = body.get("headers", {})
+        req = urllib.request.Request(url, headers=headers)
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)